src/Features/CSharp/Portable/UsePatternCombinators/CSharpUsePatternCombinatorsCodeFixProvider.cs

```diff
@@ -123,8 +128,8 @@ private static ExpressionSyntax AsExpressionSyntax(ExpressionSyntax expr, Analyz
 
         private class MyCodeAction : CustomCodeActions.DocumentChangeAction
         {
-            public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument)
-                : base(CSharpAnalyzersResources.Use_pattern_matching, createChangedDocument, nameof(CSharpUsePatternCombinatorsCodeFixProvider))
+            public MyCodeAction(string title, Func<CancellationToken, Task<Document>> createChangedDocument)
+                : base(title, createChangedDocument, nameof(CSharpUsePatternCombinatorsCodeFixProvider))
```

Comment:
I think it's better to use different equivalence keys here. The difference is when invoking a FixAll for a safe action, it doesn't apply unsafe replacements. @CyrusNajmabadi What do you think?

Reply:
You're right, I didn't think about this issue ;  however (and sorry if this a bit dumb) but how would equivalence keys help here ?

The way I would fix this is at line 75, within `FixAllAsync`, by only processing the diagnostic that have the same safety property as the current code action :

- add `private bool _isSafe` to the codefix provider class

- change `RegisterCodeFixesAsync`'s from `var isSafe = /* ... */` to `_isSafe = /* ... */`

- change `FixAllAsync`'s foreach loop to :
```csharp
foreach (var diagnostic in diagnostics.Where(d => _isSafe == d.Properties.TryGetValue("safe")) { /* ... */ }
```

I (very quickly) tested it and it seems to prevent codefixes to propagate to the wrong safe/unsafe version. Would this approach work ?

If not, I'm not sure I understand what you mean by this "equivalence key" suggestion, could you elaborate ?

Reply:
@Blokyk I haven't dug deeply into how the fix all system works in Roslyn, but it's designed that when you invoke a "FixAll" operation, it applies all available actions that have the same equivalence key (that is the third parameter passed to `base` constructor call.

Adjusting the code as follows should fix the problem:

```csharp
            public MyCodeAction(string title, Func<CancellationToken, Task<Document>> createChangedDocument, string equivalenceKey)
                : base(title, createChangedDocument, equivalenceKey)

// Call sites:
new MyCodeAction(CSharpAnalyzersResources.XXX, ....., nameof(CSharpAnalyzersResources.XXX))
```

Reply:
Yes, I tried to change the equivalence key depending on the version/safety just as you wrote here, but it didn't seem to prevent the wrong ones from being affected. It's entirely possible, however, that it's just cause I implemented it the wrong way. Would you like me to send a gist of both changes (the change you suggested and the one I described in my previous comment)

Reply:
@sharwell @CyrusNajmabadi may be able to help better here (in case I'am misunderstanding how equivalence keys work 😄 )

Reply:
Ah I see, thank you nonetheless 😄

Reply:
i'll take over here :)